homeassistant/components/roborock/vacuum.py

```diff
@@ -241,3 +264,148 @@ async def get_vacuum_current_position(self) -> ServiceResponse:
             "x": robot_position.x,
             "y": robot_position.y,
         }
+
+
+class RoborockQ7Vacuum(RoborockCoordinatedEntityB01, StateVacuumEntity):
+    """General Representation of a Roborock vacuum."""
+
+    _attr_icon = "mdi:robot-vacuum"
+    _attr_supported_features = (
+        VacuumEntityFeature.PAUSE
+        | VacuumEntityFeature.STOP
+        | VacuumEntityFeature.RETURN_HOME
+        | VacuumEntityFeature.FAN_SPEED
+        | VacuumEntityFeature.SEND_COMMAND
+        | VacuumEntityFeature.LOCATE
+        | VacuumEntityFeature.STATE
+        | VacuumEntityFeature.START
+    )
+    _attr_translation_key = DOMAIN
+    _attr_name = None
+    coordinator: RoborockB01Q7UpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: RoborockB01Q7UpdateCoordinator,
+    ) -> None:
+        """Initialize a vacuum."""
+        StateVacuumEntity.__init__(self)
+        RoborockCoordinatedEntityB01.__init__(
+            self,
+            coordinator.duid_slug,
+            coordinator,
+        )
+
+    @property
+    def fan_speed_list(self) -> list[str]:
+        """Get the list of available fan speeds."""
+        return SCWindMapping.keys()
+
+    @property
+    def activity(self) -> VacuumActivity | None:
+        """Return the status of the vacuum cleaner."""
+        assert self.coordinator.data.status is not None
```

Comment:
i believe there is a preference to avoid asserts in code. if this is for typing then either cast or add a check and deal with lowered test coverage